src/js/utils/linked-list.js

```diff
@@ -0,0 +1,84 @@
+export default class LinkedList {
+  constructor(options) {
```

Comment:
we can kill some of the conditionals in the rest of this file if we assign no-op defaults to `adoptItem` and `freeItem`:

```
constructor({adoptItem: () => {}, freeItem: () => {}}
```

And then no need to check `if this.adoptItem` in `insertBefore` or `if this.freeItem` in `remove`